src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,638 @@
+        return {device_name, AnyMap{std::pair<std::string, Any>{configs...}}};
+    }
+};
+
+/**
+ * @brief Key to pass set of property values to specified device
+ */
+static constexpr Property properties;
+
+/**
+ * @brief Read only property to get a std::string value representing a full device name.
+ */
+static constexpr Key<std::string, PropertyMutability::RO> full_name{"FULL_DEVICE_NAME"};
+
+/**
+ * @brief Read only property which defines the device architecture.
+ */
+static constexpr Key<std::string, PropertyMutability::RO> architecture{"DEVICE_ARCHITECTURE"};
+
+/**
+ * @brief Enum to define possible device types
+ */
+enum class Type {
+    INTEGRATED = 0,  //!<  Device is integrated into host system
+    DISCRETE = 1,    //!<  Device is not integrated into host system
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const Type& device_type) {
+    switch (device_type) {
+    case Type::DISCRETE:
+        return os << "discrete";
+    case Type::INTEGRATED:
+        return os << "integrated";
+    default:
+        throw ov::Exception{"Unsupported device separability type"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, Type& device_type) {
+    std::string str;
+    is >> str;
+    if (str == "discrete") {
+        device_type = Type::DISCRETE;
+    } else if (str == "integrated") {
+        device_type = Type::INTEGRATED;
+    } else {
+        throw ov::Exception{"Unsupported device type: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief Read only property to get a type of device. See Type enum definition for possible return values
+ */
+static constexpr Key<Type, PropertyMutability::RO> type{"DEVICE_TYPE"};
+
+/**
+ * @brief Read only property which defines Giga OPS per second count (GFLOPS or GIOPS) for a set of precisions supported
+ * by specified device
+ */
+static constexpr Key<std::map<ie::Precision, float>, PropertyMutability::RO> gops{"DEVICE_GOPS"};
+
+/**
+ * @brief  Read only property to get a float of device thermal
+ */
+static constexpr Key<float, PropertyMutability::RO> thermal{"DEVICE_THERMAL"};
+
+}  // namespace device
+
+/**
+ * @brief The key for enabling of dumping the topology with details of layers and details how
+ * this network would be executed on different devices to the disk in GraphViz format.
+ */
+static constexpr Key<bool, PropertyMutability::RW> dump_graph_dot{"HETERO_DUMP_GRAPH_DOT"};
+
+/**
+ * @brief The key with the list of device targets used to fallback unsupported layers
+ * by HETERO plugin
+ */
+static constexpr device::Priorities target_fallback{"TARGET_FALLBACK"};
+
+namespace execution {
+
+/**
+ * @brief The number of executor logical partitions
+ */
+static constexpr Key<int32_t, PropertyMutability::RW> streams{"STREAMS"};
+
+/**
+ * @brief Maximum number of concurent tasks executed by execututor
+ */
+static constexpr Key<int32_t, PropertyMutability::RW> concurrency{"CONCURRENCY"};
```

Comment:
is it new key w/o mapping to old API? can we remove it then, add once plugins support it?

Reply:
How will plugins support and test it if it is not represented?

Reply:
renamed